numbers aren't magic

briaguya0 opened the pull request:
did a big search for all the comboboxes using magic numbers and threw some enums in them

i also found a typo and fixed that in a few places

i also did a little logic cleanup in a couple places to make things read more clearly with the enum values

very open to naming suggestions and ideas about where these new enums should live

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh.otr.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714857.zip)
  - [soh-linux-compatibility.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714859.zip)
  - [soh-linux-performance.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714861.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714863.zip)
  - [soh-switch.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714865.zip)
  - [soh-wiiu.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714867.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/741714870.zip)
<!--- section:artifacts:end -->

Commit:
numbers aren't magic

aMannus commented on soh/soh/Enhancements/enhancementTypes.h:
```diff
@@ -0,0 +1,46 @@
+typedef enum {
```
Maybe not really related to your changes, but could be potentially rename this entire option to "Chest Appearance Matches Contents"? It feels weird to refer to it as "size" while that's only one of the available options.

garrettjoecox replied:
I usually refer to it as "Chest Style Matches Contents"

Allows it to keep it's same acronym from OOTR

briaguya0 replied:
ooh, i like style, i'll clean that up

aMannus replied:
That works too

briaguya0 replied:
https://github.com/HarbourMasters/Shipwright/pull/2976/commits/81e3ebe151a6b93b9222f1d38ee64d0f5ce2add1

aMannus commented on soh/soh/Enhancements/randomizer/randomizerTypes.h:
```diff
@@ -1764,3 +1764,30 @@ typedef struct CowIdentity {
     RandomizerCheck randomizerCheck;
 } CowIdentity;
+
+typedef enum {
```
The tracker enums feels a bit ambiguous to me, maybe specify by changing it to TRACKER_WINDOW etc instead? And WINDOW_WINDOW feels weird, maybe WINDOW_WINDOWED might be clearer?

briaguya0 replied:
how about this https://github.com/HarbourMasters/Shipwright/pull/2976/commits/3016fa79997df1663afed6394416ce31dfbd5cb5 (renamed all the usages in the next commit)

aMannus replied:
Still not a fan of TRACKER_WINDOW_WINDOW but not a blocker for me either, so up to you.

Malkierian replied:
Maybe something like TRACKER_WINTYPE_XXX instead?

briaguya0 replied:
`TRACKER_WINTYPE_WINDOW` feels as odd to me as `TRACKER_WINDOW_WINDOW`

I'm also not a huge fan of `WINDOWED` because that's more of like a windowed vs fullscreen thing in my mind than what imgui means by this

i'm happy with it as is for now and if we think of a better name later it's easy to change

briaguya0 replied:
really this problem goes beyond the enum imo, we have
![image](https://github.com/HarbourMasters/Shipwright/assets/70942617/537e29ed-684e-4a66-a37d-5e41401b62f1)

how is "Window" a window type? Also, isn't that window floating too? it definitely isn't tiled.

Malkierian replied:
Yeah, I was just trying to match it to that, lol. That is also very clunky. The problem is coming up with a pair of words more accurate and less confusing XD

aMannus replied:
Seems like something we can also address later if we want to then.

Malkierian replied:
Agreed, this PR as it stands is good to go imo.

briaguya0 replied:
https://github.com/HarbourMasters/Shipwright/issues/2978

aMannus reviewed (commented):
Very nice! Couple of suggestions from me.

Archez commented on soh/soh/Enhancements/enhancementTypes.h:
Overall, I am onboard. I wonder if we should add a little more specificity to the enum names. I'm thinking similar to how our randomizer options are all `RO_XXX`, we could for example have all the enhancement options here be prefixed with `EO_XXX`

Granted there are more enums in this PR than just this file, but just throwing out ideas to help with the namespace of things.

briaguya0 replied:
i'm down to set the pattern of `EO_` for things in this file, not sure how best to handle the things beyond this one though (maybe those can be a future problem if we can't think of something good?)

leggettc18 replied:
I disagree, I think these are fine as is since they are already being prefaced with the option they are enumerating for. I think the initial EO might just add extra noise to autocomplete stuff.

Archez replied:
My concern is less with identifying the option name, and more so with "specificity" in the include/header space to help avoid collisions, especially as this file will grow when more enhancements are added. Again just a suggestion, I won't die on the hill over it.

(The pattern is used both in the original game code for enums, and some of our own existing enums)

leggettc18 replied:
Were the enums actually in the original game code or is that decomp documentation? Anyway, I don't disagree with the specificity and pattern points, I just think that having the option name in the enums is enough to satisfy those points without the `EO` prefix. I also won't die on this hill though, adding `EO` is fine with me if bria thinks it's better.

briaguya0 replied:
i don't have strong feelings one way or the other, i just want to not have magic numbers

Malkierian replied:
I honestly don't think the prefix is necessary, either. If you really want to do specificity, then you could separate the enums into separate headers per group, but I also think the name of the option being in the enum value names is enough distinction between the enums themselves. Maybe in the enum name instead of the value names?

briaguya0 replied:
sounds like there are more people in the "don't add the EO prefix" camp at the moment. we can revisit that decision later if we want to, i'm happy without it for now

Commit:
chest style matches contents

renamed everything except the cvar itself at this point
(waiting for versioned configs for that)

Commit:
tracker as part of names

Commit:
finish the tracker rename

leggettc18 reviewed (approved):
I'm in favor of this as is, without the `EO` prefix on the enhancement option enums.
